dpypelines/pipeline/generic_file_ingress_v1.py

```diff
@@ -0,0 +1,315 @@
+    except Exception as err:
+        logger.error("Error occurred when getting Upload Service URL", err)
+        de_notifier.failure()
+        raise err
+
+    # Attempt to access the local data store
+    try:
+        local_store = LocalDirectoryStore(files_dir)
+        files_in_directory = local_store.get_file_names()
+        logger.info(
+            "Local data store created",
+            data={
+                "local_store_dir": files_dir,
+                "files_in_directory": files_in_directory,
+            },
+        )
+    except Exception as err:
+        logger.error(
+            "Error occurred when creating local data store from files directory",
+            err,
+            data={"local_store_dir": files_dir},
+        )
+        de_notifier.failure()
+        raise err
+
+    # Extract the patterns for required files from the pipeline configuration
+    try:
+        required_file_patterns = get_required_files_patterns(pipeline_config)
+        logger.info(
+            "Required file patterns retrieved from pipeline config",
+            data={"required_file_patterns": required_file_patterns},
+        )
+    except Exception as err:
+        files_in_directory = local_store.get_file_names()
+        logger.error(
+            "Error occurred when getting required file patterns",
+            err,
+            data={
+                "files_in_directory": files_in_directory,
+                "pipeline_config": pipeline_config,
+            },
+        )
+        de_notifier.failure()
+        raise err
+
+    # Check for the existence of each required file
+    for required_file in required_file_patterns:
+        try:
+            if not local_store.has_lone_file_matching(required_file):
+                email_content = file_not_found_email(required_file)
+                email_client.send(
+                    submitter_email, email_content.subject, email_content.message
+                )
+                logger.info(
+                    "Email sent to submitter about missing required file",
+                    data={
+                        "submitter_email": submitter_email,
+                        "required_file": required_file,
+                    },
+                )
+                de_notifier.failure()
+        except Exception as err:
+            files_in_directory = local_store.get_file_names()
+            logger.error(
+                "Error occurred when looking for required file",
+                err,
+                data={
+                    "required_file": required_file,
+                    "required_file_patterns": required_file_patterns,
+                    "files_in_directory": files_in_directory,
+                    "pipeline_config": pipeline_config,
+                },
+            )
+            de_notifier.failure()
+            raise err
+
+    # Extract the patterns for supplementary distributions from the pipeline configuration
```

Comment:
remove please, supplementary_distributions will be blank.

Reply:
done